src/server/game/Globals/ObjectAccessor.cpp

```diff
@@ -81,6 +81,46 @@ HashMapHolder<Player>::MapType const& ObjectAccessor::GetPlayers()
 template class TC_GAME_API HashMapHolder<Player>;
 template class TC_GAME_API HashMapHolder<Transport>;
 
+void PlayerNameMapHolder::Insert(Player* p)
+{
+    boost::unique_lock<boost::shared_mutex> lock(*GetLock());
+
+    GetContainer()[p->GetName()] = p;
+}
+
+void PlayerNameMapHolder::Remove(Player* p)
+{
+    boost::unique_lock<boost::shared_mutex> lock(*GetLock());
+
+    GetContainer().erase(p->GetName());
+}
+
+Player* PlayerNameMapHolder::Find(std::string const& name)
+{
+    std::string charName(name);
+    if (!normalizePlayerName(charName))
+        return nullptr;
+
+    {
+        boost::shared_lock<boost::shared_mutex> lock(*GetLock());
+
+        typename MapType::iterator itr = GetContainer().find(charName);
+        return (itr != GetContainer().end()) ? itr->second : nullptr;
+    }
+}
+
+auto PlayerNameMapHolder::GetContainer() -> MapType&
```

Comment:
It's a c++11 custom, to do that I'd have to scope the name MapType.

With auto keyword this is not needed as we already scoped inside PlayerNameMapHolder (return type is writen after :: )